packages/mux-uploader/src/mux-uploader.ts

```diff
@@ -257,36 +257,46 @@ class MuxUploaderElement extends globalThis.HTMLElement implements MuxUploaderEl
           : {}),
       });
 
-      this.setAttribute('upload-in-progress', '');
-
       this.dispatchEvent(
         new CustomEvent('uploadstart', { detail: { file: upload.file, chunkSize: upload.chunkSize } })
       );
+      this.setAttribute('upload-in-progress', '');
+
+      if (upload.offline) {
+        this.dispatchEvent(new CustomEvent('offline'));
+      }
 
-      upload.on('attempt', (event) => {
+      upload.on('attempt', (event: any) => {
```

Comment:
can't these be typed as the UpChunk events?

Reply:
they don't have strict types just yet (we were a bit loosey goosey in our original impl, so those didn't make the cut for my initial round of TS tightening on upchunk).